daemon/delete.go

```diff
@@ -77,9 +77,9 @@ func (daemon *Daemon) rmLink(container *container.Container, name string) error
 
 // cleanupContainer unregisters a container from the daemon, stops stats
 // collection and cleanly removes contents and metadata from the filesystem.
-func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemove, removeVolume bool) error {
+func (daemon *Daemon) cleanupContainer(container *container.Container, config *types.ContainerRmConfig) error {
 	if container.IsRunning() {
-		if !forceRemove {
+		if !config.ForceRemove {
```

Comment:
We have a new opportunity for a panic here since `config` could be nil.  Can we make `config` a `types.ContainerRmConfig` instead of a `*types.ContainerRmConfig` or alternately can we add a nil check at the top of this method?

Reply:
Good point; I had to look at my code again, wondering why I did that; looks like the existing `(daemon *Daemon) ContainerRm()` accepted a pointer (so I "punched" that through). I had a quick look, but see that specific one is part of an interface that's also implemented by BuildKit (we could probably change it, but let's do so separately).

So what I did is update just the `cleanupContainer()` to not take a pointer, which indeed prevents this from happening for the _new_ variants of passing the config struct.